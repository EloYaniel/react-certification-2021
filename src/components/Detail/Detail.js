import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { VideoPlayer } from '../Shared/VideoPlayer';
import { RelatedVideoList } from './RelatedVideoList';
import {
  Container,
  FavButton,
  MainContainer,
  SidebarContainer,
  Subtitle,
  Title,
} from './styles';
import {
  addToFavorites,
  isFavorite,
  removeFromFavorites,
} from '../../services/favorites';
import { getVideoInfo } from '../../services/youtube';

export const Detail = ({ history }) => {
  const [favorite, setFavorite] = useState(false);
  const [videoInfo, setVideoInfo] = useState({});
  const { videoID } = useParams();

  useEffect(() => {
    getVideoInfo([videoID]).then((info) => {
      setVideoInfo(info[0]);
    });
    setFavorite(isFavorite(videoID));
  }, [videoID, history]);

  const onVideoChanged = (relatedVideoID) => {
    history.push(`/detail/${relatedVideoID}`);
  };

  const handleSetFav = () => {
    if (favorite) {
      removeFromFavorites(videoID);
    } else {
      addToFavorites(videoID);
    }
    setFavorite(!favorite);
  };

  return (
    <Container>
      <MainContainer>
        <VideoPlayer videoId={videoID} />
        <FavButton onClick={handleSetFav}>
          {favorite ? 'Remove from Favorites' : 'Add to Favorites'}
        </FavButton>
        <Title>{videoInfo.title}</Title>
        <Subtitle>{videoInfo.description}</Subtitle>
      </MainContainer>
      <SidebarContainer>
        <RelatedVideoList relatedToVideoId={videoID} onVideoChanged={onVideoChanged} />
      </SidebarContainer>
    </Container>
  );
};
