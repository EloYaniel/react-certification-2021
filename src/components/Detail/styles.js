import styled from 'styled-components';

export const FavButton = styled.button`
  display:block;
  padding: 0.5rem;
  font-size: 1.5rem;
  text-align; center;
  color: #3363ff;
  background-color: #d8e0fd;
  border: none;
  border-radius: 0.4rem;
  transition: 0.2s;
  cursor: pointer;
  letter-spacing: 0.1rem;
  margin: 10px 0;

  &:hover, &:active {
    background-color: #c2cffc;
  }
`;

export const Container = styled.div`
  display: flex;
  flex-direction: row;
`;

export const MainContainer = styled.div`
  flex-grow: 4;
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: 0 30px;
`;

export const SidebarContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Title = styled.h2`
  margin: 10px;
  color: ${(props) => props.theme.textColor};
`;

export const Subtitle = styled.h4`
  color: ${(props) => props.theme.textColor};
`;
