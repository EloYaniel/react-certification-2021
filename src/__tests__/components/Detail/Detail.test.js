import React from 'react';
import { render } from '@testing-library/react';
import { Detail } from '../../../components/Detail';

describe('Detail Component tests suit', () => {
  it('Should match snapshot', () => {
    const element = render(<Detail />);
    expect(element).toMatchSnapshot();
  });
});
