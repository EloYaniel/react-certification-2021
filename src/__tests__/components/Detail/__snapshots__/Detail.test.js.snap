// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Component tests suit Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSgupP iaInnZ"
      >
        <div
          class="sc-gKsewC cbsqwP"
        >
          <iframe
            class="sc-bdfBwQ gTSScc"
            src="https://www.youtube-nocookie.com/embed/undefined"
          />
          <button
            class="sc-eCssSg hDweQF"
          >
            Add to Favorites
          </button>
          <h2
            class="sc-fubCfw hORWeU"
          />
          <h4
            class="sc-pFZIQ jZnnif"
          />
        </div>
        <div
          class="sc-iBPRYJ jHiIsc"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSgupP iaInnZ"
    >
      <div
        class="sc-gKsewC cbsqwP"
      >
        <iframe
          class="sc-bdfBwQ gTSScc"
          src="https://www.youtube-nocookie.com/embed/undefined"
        />
        <button
          class="sc-eCssSg hDweQF"
        >
          Add to Favorites
        </button>
        <h2
          class="sc-fubCfw hORWeU"
        />
        <h4
          class="sc-pFZIQ jZnnif"
        />
      </div>
      <div
        class="sc-iBPRYJ jHiIsc"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
